Formula/mysql@5.6.rb

```diff
@@ -22,7 +22,19 @@ class MysqlAT56 < Formula
   depends_on "openssl@1.1"
 
   def datadir
-    var/"mysql"
+    return var/"mysql" if default_datadir_already_in_use_by_this_version?
+
+    var/"mysql56"
+  end
+
+  # TODO: Could also try running file on files in var/*.frm if they exist
+  # e.g.
+  # peter@Peters-iMac homebrew-core % file /usr/local/var/mysql56/mysql/columns_priv.frm
+  # /usr/local/var/mysql56/mysql/columns_priv.frm: MySQL table definition file Version 9,
+  # type MYISAM, MySQL version 50647
+  #
+  def default_datadir_already_in_use_by_this_version?
+    (var/"mysql").exist? && var.glob("mysql/*.dblwr").empty? && (var/"mysql/test").exist?
```

Comment:
Think it's enough to just check for `(var/"mysql").exist?`?

Reply:
Could also output something in the `caveats` based on this?

Reply:
Sounds good,  I think that a simpler check and a caveat explaining the location of the datadir to the user would be preferable to trying to work it out automatically for them. I'll modify as you suggest.

Reply:
Update: It turns out this is harder to do than I expected, as putting a conditional in the `datadir` method means it gets evaluated differently during various stages in the formula.